prober/http.go

```diff
@@ -226,6 +228,28 @@ func (t *transport) GotFirstResponseByte() {
 	t.current.responseStart = time.Now()
 }
 
+// chainedReadCloser implementes an io.ReadCloser that refers to two io.ReadCloser instances, forwarding all Read
```

Comment:
A reader that counted the bytes that passed through it would seem a bit simpler than a unusual tee.

Reply:
The intention behind that code is to be able to read from the gzip Reader, but at the same time be able to close both Closer's, if they exist. The original body needs to be closed to be able to reuse the TCP connection and free the file descriptor; the gzip reader needs to be closed in order to check if there are decompression errors.

I think your comment refers to the use of io.TeeReader in a different part of the change. I'll fix that, because it's true that I'm abusing the fact that `matchRegularExpressions` is using ioutil.ReadAll.

Reply:
OK, removed.